### PR TITLE
Support git-bash in Windows.

### PR DESCRIPTION
**Pull request by ipcjs**

### CODE REVIEW

**Comment by jonsmithers** [comment]
This looks really cool! I will look at this tomorrow evening.

File: package.json [hunk 1]
```diff
@@ -54,6 +54,11 @@
                     "default": false,
                     "description": "(EXPERIMENTAL) Sync cursor position to vscode when vim quits"
                 },
+                "open-in-vim.integratedShellPath": {
+                    "type": "string",
+                    "default": null,
+                    "definition": "Shell path of integrated terminal."
+                },
```
Comment (jonsmithers) [inline package.json]: I wonder if this should be removed. Why would a user want to change `integratedShellPath`?

Comment (ipcjs) [reply package.json]: On windows, we can install git to other directories.😅
ScreenClip

Comment (jonsmithers) [reply package.json]: That makes sense. Good point :smile:

**Comment by jonsmithers** [comment]
I noticed that there are actually two `bash.exe`s on my system: 
* `C:\Program Files\Git\bin\bash.exe`
* `C:\Windows\System32\bash.exe`.

`Git\bin\bash.exe` sees the `C:\` drive at `/c/`.
`System32\bash.exe` sees the `C:\` drive at `/mnt/c/`. 

It seems like `System32\bash.exe` is for a WSL I installed (Windows Subsystem for Linux).

I think `C:\Program Files\Git\bin\bash.exe` is more likely to exist on users' machines, but right now I'm seeing strange errors when running vim from Git\bin\bash.exe inside vscode. I don't know what changed!

```
--- Autocommands ---
Error detected while processing /etc/vimrc:
line   43:
E10: \ should be followed by /, ? or &
line   44:
E10: \ should be followed by /, ? or &
line   45:
E10: \ should be followed by /, ? or &
line   46:
E10: \ should be followed by /, ? or &
line   47:
E10: \ should be followed by /, ? or &
line   48:
E10: \ should be followed by /, ? or &
line   53:
E10: \ should be followed by /, ? or &
line   54:
E10: \ should be followed by /, ? or &
Error detected while processing function plug#end:
line   86:
E484: Can't open file C:Program Files (x86)Vimvim81/syntax/syntax.vim
Error detected while processing /c/Users/JonSm/_vimrc:
line  981:
E484: Can't open file C:Program Files (x86)Vimvim81/defaults.vim
Press ENTER or type command to continue
```

`system32\bash.exe` works fine for me (after fixing filepaths to use `/mnt/c` instead of `/c`).

**Comment by jonsmithers** [comment]
I restarted vscode and those errors stopped. ¯\\\_(ツ)\_/¯

**Comment by ipcjs** [comment]
Maybe we should also support Windows Subsystem for Linux?

**Comment by jonsmithers** [comment]
It's a little difficult to support both. It seems like we'd have to launch a separate shell instance just to check where the `C:` drive is (in order to conditionally update _all_ paths to have a `/mnt` prefix).

```bash
test -d /mnt/c
exit $?
```

I also encountered some colorscheme issues when using WSL, but I didn't look into why.

For now, maybe we should only support git bash.exe. Maybe this will avoid unnecessary complexity.

**Pull request merged**